site/docs/security/sasl.md

```diff
@@ -0,0 +1,197 @@
+        };
+
+
+
+    The `Bookie` section in the JAAS file tells the bookie which principal to use and the location of the keytab where this principal is stored.
+    It allows the bookie to login using the keytab specified in this section. See [notes](#notes) for more details on Zookeeper’s SASL configuration.
+
+2. Pass the name of the JAAS file as a JVM parameter to each Bookie:
+
+        -Djava.security.auth.login.config=/etc/bookkeeper/bookie_jaas.conf
+
+
+    You may also wish to specify the path to the `krb5.conf` file
+    (see [JDK’s Kerberos Requirements](https://docs.oracle.com/javase/8/docs/technotes/guides/security/jgss/tutorials/KerberosReq.html) for more details):
+
+
+        -Djava.security.krb5.conf=/etc/bookkeeper/krb5.conf
+
+
+3. Make sure the keytabs configured in the JAAS file are readable by the operating system user who is starting the Bookies.
+
+4. Enable SASL authentication plugin in the bookies by setting following parameters.
+
+
+        bookieAuthProviderFactoryClass=org.apache.bookkeeper.sasl.SASLBookieAuthProviderFactory
+        # if you run `autorecovery` along with bookies
+        clientAuthProviderFactoryClass=org.apache.bookkeeper.sasl.SASLClientProviderFactory
+
+
+### Configuring Clients
+
+To configure SASL authentication on the clients:
+
+1. Clients will authenticate to the cluster with their own principal (usually with the same name as the user running the client),
+    so obtain or create these principals as needed. Then create a `JAAS` file for each principal. The `BookKeeper` section describes
+    how the clients like writers and readers can connect to the Bookies. The following is an example configuration for a client using
+    a keytab (recommended for long-running processes):
+
+
+        BookKeeper {
+            com.sun.security.auth.module.Krb5LoginModule required
+            useKeyTab=true
+            storeKey=true
+            keyTab="/etc/security/keytabs/bookkeeper.keytab"
+            principal="bookkeeper-client-1@EXAMPLE.COM";
+        };
+
+
+2. Pass the name of the JAAS file as a JVM parameter to the client JVM:
+
+        -Djava.security.auth.login.config=/etc/bookkeeper/bookkeeper_jaas.conf
+
+
+    You may also wish to specify the path to the `krb5.conf` file (see
+    [JDK’s Kerberos Requirements](https://docs.oracle.com/javase/8/docs/technotes/guides/security/jgss/tutorials/KerberosReq.html) for more details).
+
+
+        -Djava.security.krb5.conf=/etc/bookkeeper/krb5.conf
+
+
+3. Make sure the keytabs configured in the `bookkeeper_jaas.conf` are readable by the operating system user who is starting bookkeeper client.
+
+4. Enable SASL authentication plugin in the client by setting following parameters.
+
+        clientAuthProviderFactoryClass=org.apache.bookkeeper.sasl.SASLClientProviderFactory
+
+
+## Enabling Logging for SASL
+
+To enable SASL debug output, you can set `sun.security.krb5.debug` system property to `true`.
```

Comment:
Java GSSAPI errors are not clear, most of the times I need to enable debug to understand the cause.
in my cases most of the times the problems are related to reverse-lookup of IP addresses to hostname.
for Kerberos it is very important that DNS is setup coherently both for names and for PTRs

there is also a 'debug=true' option in Krb5LoginModule
```
Server {
  com.sun.security.auth.module.Krb5LoginModule required debug=true
  useKeyTab=true
  keyTab="xxxxx.keytab"
  storeKey=true
  useTicketCache=false
  principal="xxx/xxx@xxx";
};
```